platform/util/platform-util/src/main/java/org/codice/ddf/platform/util/TimeLimitedMatcher.java

```diff
@@ -0,0 +1,171 @@
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.platform.util;
+
+import java.time.Duration;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Customized and reduced-scope RegEx matcher class to protect against runaway searches.
+ *
+ * <p>Instances of this class, created through one of its two factory methods, will either succeed
+ * or timeout, throwing a RuntimeException - specifically {@link TimeoutException} - in the latter
+ * case. It is incumbent on the caller to handle the runtime exception.
+ *
+ * <p>Example usage:
+ *
+ * <pre>
+ *   try {
+ *     Pattern pattern = Pattern.compile("bc");
+ *     TimeLimitedMatcher matcher = TimeLimitedMatcher.create(pattern, "abc123");
+ *     if (matcher.matches()) {
+ *       // do something with the match
+ *     }
+ *   } catch (TimeLimitedMatcher.TimeoutException e) {
+ *     // handle exception
+ *   }
+ * </pre>
+ */
+public class TimeLimitedMatcher {
+  // Limit the lifetime of this sequence to five seconds by default.
+  private static final Duration DEFAULT_DURATION = Duration.ofSeconds(5L);
+
+  private final Matcher matcher;
+
+  private boolean matched;
+
+  /**
+   * Creates a new Time-limited matcher instance to search for the regular expression {@code
+   * pattern} against the {@code inputString}.
+   *
+   * <p>This method executes the search before returning, throwing a {@link TimeoutException} if the
+   * regex search runs longer than {@link #DEFAULT_DURATION}.
+   *
+   * @param pattern RegEx search pattern
+   * @param inputString String to match against.
+   * @return TimeLimitedMatcher instance to use for reporting out match details
+   */
+  public static TimeLimitedMatcher create(Pattern pattern, CharSequence inputString) {
+    return create(pattern, inputString, DEFAULT_DURATION);
+  }
+
+  /**
+   * Creates a new Time-limited matcher instance to search for the regular expression {@code
+   * pattern} against the {@code inputString}.
+   *
+   * <p>This method executes the search before returning, throwing a {@link TimeoutException} if the
+   * regex search runs longer than {@code duration}.
+   *
+   * @param pattern RegEx search pattern
+   * @param inputString String to match against.
+   * @param duration the length of time to continue processing the regex before failing
+   * @return TimeLimitedMatcher instance to use for reporting out match details
+   */
```

Comment:
✏️ Should document that the timeout exception can be thrown here. Although, my preferences would be to not auto-match here but let the client call `match()` to start the process. This would allow pre-recreating matchers before using them.

Reply:
I did indicate the exception in the body of the javadoc and feel a bit dirty about adding an `@throws` doc line for a runtime exception.

Reply:
Don't add a throws, just javadoc.